### ISSUE TITLE
What am I looking at exactly?

### ISSUE THREAD

**Issue by Joz3d** [mempool/mempool#421]
I feel like I'm fairly knowledgeable about Bitcoin.... and I love mempool.space and the fee estimation and block exploring feature of it as well, but the core mempool number itself I'm confused about.  For example, currently (on the official site) it's showing ~200 MB in the mempool as I would understand it (in the Brown blocks on the left)... on blockchain.com and Johoe's it has the mempool size as ~80 MB.  On my own node it shows 'bytes' as ~80 MB as well.  It shows 'usage' however at ~280 MB, which is more than your 200MB to make matters worse.  So I still don't know what the heck the size of the mempool is!?  It seems that the majority of sites are saying 80 MB, but mempool.space says ~200 MB.  Then lower down under 'Memory usage' it says "455 MB / 300 MB"?  ...what??  That sounds like default node maxmempool size of 300 MB, but you must have your node configured to store more than that?  So why would it still be showing it out of 300 MB?  And why is mine saying 280 MB and not 300 if it's higher? :confused:

What I'm getting at here is that I can't find an explanation of what the heck is going on.  There are some nice mouse-overs for parts of mempool.space but other areas such as the upcoming blocks don't have any and I feel they desperately need an explanation.  I think there are many non-mempool experts who would like to understand what this ~200 MB is and why it's reporting so much higher than other sites?  Is this explained anywhere as I feel that I've really tried to find the answer without success. So more explanation/mouse-overs is my report here but also if you could explain to me the discrepancy as it's driving me nuts! :wink:  Meaning the 80 MB vs. 200 MB vs. 455 MB!

**Comment by softsimon** [comment]
Good question. Will look into the different numbers and try make sense of it later :)

In the meantime maybe you'll find something useful here https://bitcoinbriefly.com/how-to-use-mempool-space-block-explorer/

**Comment by Joz3d** [comment]
Thanks, that is a fantastic doc that I would highly recommend linking from the front page.  Very informative, but as you alluded, I didn't see anything in there unfortunately to explain how that last block's predicted size (mempool size?) is calculated.  It did confirm to me though that the Memory Usage 300 MB is the default node max... which I wonder if there is a point of displaying it anymore if whatever that aspect of the mempool is never comes back down.

**Comment by wiz** [comment]
Well, the concept of a "mempool" is a very deep rabbit hole... I'm not sure it's possible to explain everything just using mouseovers, and you really do need to learn about Bitcoin to some extent before you can use the site. For example, you could ask similar questions about what is a block explorer, what are inputs and outputs, what are transaction fees, what is segwit, etc. etc. etc.

Of course there's always room for improvement and tweaking our design to be more intuitive, but if we added mouseovers for every single thing it would kinda ruin the clean design of the site. As for the different numbers, I suppose you need to understand MvB vs MB and how SegWit and Weight Units work, and also how Bitcoin Core stores the mempool transactions in actual memory, and what overhead is involved for that - the short answer is: to store 200 MB (135 MvB) of transactions in memory, Bitcoin Core uses 458 MB of RAM on the actual server due to the overhead required, and of course our mempool is not "the" mempool, every site will be different and could have its own configuration, even our individual servers have slightly varying mempool numbers. So yeah, it's not so simple 😅

**Comment by Joz3d** [comment]
Right, I get what you're saying with the audience and I'm aware of the Segwit change to weight units, but I probably need to study  how that corresponds to vBytes more and where which is used... though I thought vBytes would always be higher than corresponding regular Bytes?

The 458 MB RAM consumption makes sense.. I would say maybe even call it _RAM_ or _RSS Usage_ or _Node Memory Used_ for clarity, but just my 2sats.

Anyway, I'm not suggesting mouseover on everything (I'm actually a minimalist) but when you have the other big sites reporting a mempool of 80 MB and over here we're reporting 200 MB... it kinda begs an explanation.

I'm sure I'm not the only person who is wondering what is going on.  I'm guessing most visitors that saw the lower mempool size prior to coming here are either going to be thinking:
a) "Wow that went up fast!"
b) "Huh what?  I guess I don't get it and have no idea what the mempool size is... somewhere between 80 and 200! :shrug: "

**Comment by wiz** [comment]
And just think, if you install mempool on your own hardware at home then you can have 3 numbers to consider 😉

**Comment by Joz3d** [comment]
I was actually trying that prior to posting this question to investigate what my instance would reveal in brown block sizes and try to figure it out... but am running it on Lubuntu and spent an hour resolving MariaDB auth issues with the way Ubuntu auth is used with that... and by the time I got that working I realized I missed the electrum server requirement (really? on a full node? [I'm sure you have a good reason, probably some better APIs]) and forgot I still had no nginx and lost steam. :dash:
